common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -93,17 +93,48 @@
     <%doc>
         Loads a studio-frontend page, with the necessary context. Context is expected
         as a dictionary in the body of this tag.
-
-        Dev note: we could also add the locale-injection script in this block
-            -use a better default than the hardcoded 'en'. There should be a setting or something?
-            -lookup (webpack exported) locale-injection script using lang as key
-            -include it as the first script in this block
     </%doc>
     <%
         body = capture(caller.body)
+        body_dict = json.loads(body)
+        locale = body_dict['lang']
+        messages = "{}"
+        localeData = ""
+
+        if locale != 'en':
+            # because en is the default, studio-frontend will have it loaded by default
+            ietf_to_transifex = {
+                'fr': 'fr_CA',
```

Comment:
Could you explain why this dict is necessary? Is the locale that's being passed into the body not quite what we need? Is there a way to tweak the Python code to pass through the correct data? It looks like it's replacing hyphens with underscores and appending Canada to the French language. I'm wondering why that's necessary.

Reply:
Yeah, it's massaging data formats - the "correct" tweak in python-land will be similar to what I've done here.

First, the hyphen-to-underscore: django correctly uses language codes as referenced in [RFC 5646](https://tools.ietf.org/html/rfc5646), whereas Transifex ever-so-helpfully returns [translated po files using underscores in the tag](https://github.com/edx/studio-frontend/tree/master/src/data/i18n/locales/po). We could move the logic into SFE's post-processing of the po files if we wanted, but the RFC5646/Transifex mismatch will need to be resolved somewhere.

The french canadian thing is much simpler - we have [0 translated strings for *plain* french](https://github.com/edx/studio-frontend/blob/master/src/data/i18n/locales/json/fr.json), but [all the strings for *Canadian* french](https://github.com/edx/studio-frontend/blob/master/src/data/i18n/locales/json/fr_CA.json). I decided that it would be better for a French speaking user to see some French rather than none, and also none of the other languages we support were fully translated when I started manually testing this.

So I need to ping Shelby/Bill on the french canadian issue, hopefully we can find someone to sign off on copying the `fr_CA` translations over to just `fr` in Transifex and remove that piece of specialized logic. Once that's done I'll remove the dict and replace it with a simpler `locale.replace('_', '-')`.

Reply:
Offline update: I spoke with @sstack22, I'm going to manually update Transifex to use the current `fr-CA` translations for `fr`